src/types/timestamp.rs

```diff
@@ -1176,32 +1233,36 @@ mod timestamp_tests {
             .with_hour_and_minute(17, 42)
             .build_at_offset(5 * 60)?;
         assert_ne!(timestamp1, timestamp2);
+        assert!(!timestamp1.ion_eq(&timestamp2));
         Ok(())
     }
 
     #[test]
     fn test_timestamps_with_different_days_are_not_equal() -> IonResult<()> {
         let builder = Timestamp::with_year(2021).with_month(2);
-        let timestamp1 = builder.clone().with_day(5).build();
-        let timestamp2 = builder.with_day(6).build();
+        let timestamp1 = builder.clone().with_day(5).build()?;
+        let timestamp2 = builder.with_day(6).build()?;
```

Comment:
🗺️ A few of these unit tests accidentally forgot to unwrap the `Result<_>` using `?`. This worked when using `assert_eq` because it can compare two `Result` values via `PartialEq`. However, when I added calls to `ion_eq` it rightly complained that `IonEq` is not implemented for `Result`.